src/components/Pokedex.tsx

```diff
@@ -1,9 +1,17 @@
 import { StatusBar } from 'expo-status-bar';
-import { StyleSheet, Text, TouchableHighlight, View } from 'react-native';
+import { ActivityIndicator, Button, SafeAreaView, StyleSheet, Text, TouchableHighlight, View } from 'react-native';
 import { FlatList } from 'react-native';
+import { useInfiniteQuery } from '@tanstack/react-query'
+import { Fragment } from 'react';
+import queryString from 'query-string';
+
+
+type Pokemon = { name: string, url: string };
+
+type QueryData = { count: number, next: string, previous: string, results: Pokemon[] };
```

Comment:
Would extract these into a types.d.ts file. Would also try to name QueryData something more specific to the query it is calling. It is very very likely that this will be needed in other files too on a larger project so makes it much cleaner to have it in a types file 

The prop types can generally stay in the same file though